File: skills/investgame-format/brand-kit/assets/ig_helpers.py
"""InvestGame brand constants for programmatic builds (PowerPoint / Excel).

Import these instead of re-typing hex codes - referring to colours by name is
how you keep a deck on-brand and avoid silent drift. Hex values are stored
WITHOUT the leading '#'; python-pptx wants RGBColor(0x..), pptxgenjs wants the
bare 6-char string.

Two skins share one teal/navy core:
  IG_DARK   - Dark Navy deck   (slide 13.33in x 7.5in / 16:9)
  IG_WHITE  - Warm White deck   (slide 13.33in x 7.5in / 16:9)
Both carry the InvestGame logo. See references/pptx-and-pdf.md for the
copy-the-template workflow.
"""

# --- Shared brand core --------------------------------------------------------
IG = {
    "teal": "61BFB3",  # PRIMARY brand teal - both skins
    "teal_bright": "1ECABA",
    "teal_deep": "00928A",
    "blue": "6189D6",
    "blue_deep": "2D5BA8",
    "white": "FFFFFF",
}

# --- Dark Navy skin -----------------------------------------------------------
IG_DARK = {
    "navy": "0E1F33",  # primary dark bg + body text on light
    "navy_2": "142A44",
    "teal": "61BFB3",
    "teal_2": "1ECABA",  # big numerals on dark
    "teal_dark": "00928A",  # chips, emphasis, positive delta
    "blue": "6189D6",
    "blue_dark": "2D5BA8",
    "slate": "8A95A4",
    "gray": "8FA1B5",  # labels, captions, footer
    "light": "F5F7FA",
    "rule": "E1E6EC",
    "ink_soft": "45596F",  # secondary body text on light
    "tint_teal": "E1F4F1",  # teal-tinted row / highlight
}

# --- Warm White skin ----------------------------------------------------------
IG_WHITE = {
    "bg": "F4F3EE",  # warm off-white page
    "bg_soft": "EEEDE6",
    "card": "FFFFFF",
    "text": "0B1A2A",  # warm navy body text
    "text_muted": "5F6B7A",
    "text_soft": "8E99A6",
    "accent": "61BFB3",  # primary teal
    "accent_deep": "3D9B8F",  # kickers, footnote marks
    "accent_soft": "E6F4F2",  # teal-tinted callout / row
    "warn": "C07B5A",  # rust caution / negative
    "warn_soft": "F5E6DD",
    "border": "E0DFD9",
    "border_strong": "C7C6C0",
}

# --- Chart series order (apply 1..n in this sequence) -------------------------
# Positive delta = teal_deep. Negative / caution = warn (rust). Never green;
# never red except the rust caution tone. See references/charts.md.
CHART_COLORS = ["61BFB3", "6189D6", "00928A", "2D5BA8", "8A95A4", "1ECABA", "8FA1B5"]

POSITIVE = "00928A"  # teal-deep - "up"
NEGATIVE = "C07B5A"  # rust - "down" / caution

# --- Typography ---------------------------------------------------------------
# Dark Navy skin:   Helvetica Neue (Arial fallback) throughout.
# Warm White skin:  Space Grotesk -> headings/cover; Inter -> body;
#                   JetBrains Mono -> numeric columns. PowerPoint fallbacks:
#                   Space Grotesk -> Arial Bold; Inter -> Arial / Aptos;
#                   JetBrains Mono -> Consolas. See references/typography.md.
FONTS_DARK = {"display": "Helvetica Neue", "body": "Helvetica Neue", "fallback": "Arial"}
FONTS_WHITE = {"display": "Space Grotesk", "body": "Inter", "mono": "JetBrains Mono"}


def hex_to_rgbcolor(hex6):
    """'61BFB3' -> RGBColor for python-pptx. Usage:
    from pptx.dml.color import RGBColor
    RGBColor(*hex_to_rgbcolor(IG['teal']))
    """
    return tuple(int(hex6[i : i + 2], 16) for i in (0, 2, 4))

File: skills/investgame-format/brand-kit/assets/ig_helpers.js
/* InvestGame brand constants + Chart.js defaults for HTML/PDF decks.
 *
 * Refer to colours by name so artifacts stay on-brand and never drift to a
 * hand-picked hex. Two skins share one teal/navy core. See references/charts.md
 * for chart recipes and references/colors.md for the full palette rationale. */

const IG = {
  // shared brand core
  teal: "#61BFB3", tealBright: "#1ECABA", tealDeep: "#00928A",
  blue: "#6189D6", blueDeep: "#2D5BA8", white: "#FFFFFF",

  // Dark Navy skin
  dark: {
    navy: "#0E1F33", navy2: "#142A44",
    teal: "#61BFB3", teal2: "#1ECABA", tealDark: "#00928A",
    blue: "#6189D6", blueDark: "#2D5BA8",
    slate: "#8A95A4", gray: "#8FA1B5",
    light: "#F5F7FA", rule: "#E1E6EC", inkSoft: "#45596F", tintTeal: "#E1F4F1",
  },

  // Warm White skin
  white_skin: {
    bg: "#F4F3EE", bgSoft: "#EEEDE6", card: "#FFFFFF",
    text: "#0B1A2A", textMuted: "#5F6B7A", textSoft: "#8E99A6",
    accent: "#61BFB3", accentDeep: "#3D9B8F", accentSoft: "#E6F4F2",
    warn: "#C07B5A", warnSoft: "#F5E6DD",
    border: "#E0DFD9", borderStrong: "#C7C6C0",
  },

  // Apply chart series in this order. Positive = tealDeep, negative/caution = rust.
  // Never green; never red except the rust caution tone.
  chartColors: ["#61BFB3", "#6189D6", "#00928A", "#2D5BA8", "#8A95A4", "#1ECABA", "#8FA1B5"],
  positive: "#00928A",
  negative: "#C07B5A",
};

/* Chart.js global defaults. Call once after Chart.js loads, before creating
 * charts. Pass dark:true for the Dark Navy skin. Tooltips and hover are ON by
 * default in Chart.js - keep them, the on-screen deck is meant to be explored.
 * IMPORTANT: every Chart.js chart on a slide that is shown/hidden must call
 * chart.resize() when its slide becomes visible (and on beforeprint) or it
 * renders at 0px on first paint / exports blank to PDF. */
function igChartDefaults(Chart, { dark = false } = {}) {
  const skin = dark ? IG.dark : IG.white_skin;
  const muted = dark ? "rgba(255,255,255,0.6)" : skin.textMuted;
  const grid = dark ? "rgba(255,255,255,0.12)" : "rgba(11,26,42,0.08)";
  Chart.defaults.font.family = "Inter, sans-serif";
  Chart.defaults.font.size = 12;
  Chart.defaults.color = muted;
  Chart.defaults.plugins.legend.labels.boxWidth = 12;
  Chart.defaults.plugins.legend.labels.boxHeight = 12;
  // On-brand, readable hover tooltips.
  Chart.defaults.plugins.tooltip.backgroundColor = dark ? "#0E1F33" : "#0B1A2A";
  Chart.defaults.plugins.tooltip.titleColor = "#FFFFFF";
  Chart.defaults.plugins.tooltip.bodyColor = "#FFFFFF";
  Chart.defaults.plugins.tooltip.borderColor = IG.teal;
  Chart.defaults.plugins.tooltip.borderWidth = 1;
  Chart.defaults.plugins.tooltip.padding = 10;
  Chart.defaults.interaction = { mode: "index", intersect: false };
  Chart.defaults.scale.grid.color = grid;
  Chart.defaults.scale.grid.drawTicks = false;
  Chart.defaults.scale.border.display = false;
  return { muted, grid };
}

if (typeof module !== "undefined") module.exports = { IG, igChartDefaults };
